ui/admin/app/controllers/scopes/scope/workers/worker/tags.js

```diff
@@ -44,6 +47,34 @@ export default class ScopesScopeWorkersWorkerTagsController extends Controller {
     );
   }
 
+  updateApiTags() {
+    const apiTags = structuredClone(this.model.api_tags);
+    const { key, value } = this.modalTag;
+    const editedValues = this.editValue.split(',').map((value) => value.trim());
+
+    // remove the old value
+    apiTags[key] = apiTags[key].filter((val) => val !== value);
+
+    // remove key if value is empty array
+    if (apiTags[key].length === 0) {
+      delete apiTags[key];
+    }
+
+    // add new key/value to apiTags
+    if (this.editKey in apiTags) {
+      // if editKey exist, add the new values to its array if not already preset
+      editedValues.forEach((value) => {
+        if (!apiTags[this.editKey].includes(value)) {
+          apiTags[this.editKey].push(value);
+        }
```

Comment:
Rather than doing a check for each value, isn't it more efficient to just combine the arrays and make the array unique after?

Reply:
It sure is. Thanks for the suggestion.